client/store/category.js

```diff
@@ -18,21 +15,32 @@ const initialState = {
   popularCategories: [],
 }
 
-
 /**
  * ACTION CREATORS
  */
+const getAllParentCategories = (categories) => {
+  return {
+    type: GET_ALL_PARENT_CATEGORIES,
+    categories
+  }
+}
+
 const setPopularCategoriesAC = (popularCategories) => {
   return {
     type: SET_POPULAR_CATEGORIES,
     popularCategories,
   }
 }
 
-
 /**
  * THUNK CREATORS
  */
+
+export const getAllParentCategoriesThunk = () => {
+  return async (dispatch) => {
+    const { data } = await axios.get('/api/categories/all/parent')
+    dispatch(getAllParentCategories(data))
+
```

Comment:
this function needs 2 ending curly brackets. - 
1 for thunk
1 for the async(dispatch)